dev-tools/mage/crossbuild.go

```diff
@@ -182,32 +182,6 @@ func CrossBuild(options ...CrossBuildOption) error {
 	// Each build runs in parallel.
 	Parallel(deps...)
 
-	// // It needs to run after all the builds, as it needs the darwin binaries.
-	// if err := assembleDarwinUniversal(params); err != nil {
-	// 	return err
-	// }
-
-	return nil
-}
-
-// assembleDarwinUniversal checks if darwin/amd64 and darwin/arm64 were build,
-// if so, it generates a darwin/universal binary that is the merge fo them two.
-func assembleDarwinUniversal(params crossBuildParams) error {
-	if IsDarwinUniversal() {
-		builder := GolangCrossBuilder{
-			// the docker image for darwin/arm64 is the one capable of merging the binaries.
-			Platform:      "darwin/arm64",
-			Target:        "assembleDarwinUniversal",
-			InDir:         params.InDir,
-			ImageSelector: params.ImageSelector}
-		if err := builder.Build(); err != nil {
-			return errors.Wrapf(err,
-				"failed merging darwin/amd64 and darwin/arm64 into darwin/universal target=%v for platform=%v",
-				builder.Target,
-				builder.Platform)
-		}
-	}
```

Comment:
[Blocker]
For the same reason as above, either keep it here or port it to elastic-agent-libs